spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java

```diff
@@ -252,6 +251,37 @@ private void deployStream(StreamDefinition stream, Map<String, String> cumulated
 		}
 	}
 
+	private ModuleType determineModuleType(ModuleDefinition moduleDefinition, int index, boolean hasNext) {
```

Comment:
> because we will want to support more flexible topologies

I was expecting that answer, fine by me.

> you could rely on the info in the ModuleRegistration

The problem is exactly that to get hold of a `ModuleRegistration`, I need a type :)


`isFirst` is good, and would shift the weirdness at the point where it is computed, rather than where it is used.